pkg/retry/retry.go

```diff
@@ -54,7 +54,7 @@ func Do(fn func() error, opts ...Option) error {
 			return nil
 		}
 
-		if cfg.maxAttempts != 0 && attempts >= cfg.maxAttempts {
+		if cfg.maxAttempts != 0 && attempts > cfg.maxAttempts {
```

Comment:
nit: if this behavior was unexpected, should we write a test?

Reply:
+1, am all ears. It was unexpected for me for `fn` to execute two times when setting `WithMaxAttempts(3)`.